### DISCUSSION TITLE
Could not launch module dreamview

### DISCUSSION THREAD

**Discussion by i-kj** [guardstrikelab/carla_apollo_bridge#124] [Q&A] [answered]
Hello all,
I am facing the below error when i try to start Dreamviewer from the container. I also previously encountered GPU could not be accessed warning. Since it was a warning I went a head with my installation thinking CPU is enough. 

**root@in-dev-docker:**/apollo# ./scripts/bootstrap.sh
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
nohup: appending output to 'nohup.out'
nohup: failed to run command 'cyber_launch': No such file or directory
[ERROR] Could not launch module monitor. Is it already built?
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
nohup: appending output to 'nohup.out'
nohup: failed to run command 'cyber_launch': No such file or directory
[ERROR] Could not launch module dreamview. Is it already built?
root@in-dev-docker:/apollo#

I am relatively new to this. Any help would be really appreciated and I can provide more information if needed. Thanks in advance.

**Comment by XiaoFei9704** [comment]
It seems that you did not compile successfully, you run Apollo.sh build and see if there are any errors?

**Reply by i-kj** [reply]
_I managed to successfully execute dev_start.sh and received the below message:_

[ OK ] Congratulations! You have successfully finished setting up Apollo Dev Environment.
[ OK ] To login into the newly created apollo_dev_root container, please run the following command:
[ OK ]   bash docker/scripts/dev_into.sh
[ OK ] Enjoy!


_than I used sudo to bash dev_into.sh_

BUSINESSline:~/apollo$ sudo bash docker/scripts/dev_into.sh
root@in-dev-docker:/apollo# 

 
_and when I tried making the GPU version apollo.sh i got the below message/error:_ 

root@in-dev-docker:/apollo# ./apollo.sh build_gpu 

WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[INFO] Apollo Environment Settings:
[INFO]     APOLLO_ROOT_DIR: /apollo
[INFO]     APOLLO_CACHE_DIR: /apollo/.cache
[INFO]     APOLLO_IN_DOCKER: true
[INFO]     APOLLO_VERSION: guardstrike/apollo_carla_bridge-2022-11-23-653b3e953
[INFO]     DOCKER_IMG: dev-x86_64-18.04-20210914_1336
[INFO]     APOLLO_ENV:  STAGE=dev USE_ESD_CAN=false
[INFO]     USE_GPU: USE_GPU_HOST=0 USE_GPU_TARGET=0
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[INFO] Configure .apollo.bazelrc in non-interactive mode
You have bazel 3.7.1 installed.
Found possible Python library paths:
  /usr/lib/python3/dist-packages
  /usr/local/lib/python3.6/dist-packages
Found CUDA 11.1 in:
    /usr/local/cuda-11.1/targets/x86_64-linux/lib
    /usr/local/cuda-11.1/targets/x86_64-linux/include
Found cuDNN 8 in:
    /usr/lib/x86_64-linux-gnu
    /usr/include
Found TensorRT 7 in:
    /usr/lib/x86_64-linux-gnu
    /usr/include/x86_64-linux-gnu

**Right now stuck at this step:** 
[ OK ] Successfully configured .apollo.bazelrc in non-interactive mode.
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] USE_GPU=1 without GPU can't compile. Exiting ...
root@in-dev-docker:/apollo# ./scripts/bootstrap.sh
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
nohup: appending output to 'nohup.out'
nohup: failed to run command 'cyber_launch': No such file or directory
[ERROR] Could not launch module monitor. Is it already built?
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
nohup: appending output to 'nohup.out'
nohup: failed to run command 'cyber_launch': No such file or directory
[ERROR] Could not launch module dreamview. Is it already built?


_I am still struggling to figure out the issue. Other than facing the GPU warning in my previous steps, I performed all the steps successfully._ 

**Warning message:**
Digest: sha256:4c8749787adc30de8d64321589adfde5905c45fc22cfc5aee435e2ca5d20733f
Status: Downloaded newer image for apolloauto/apollo:dev-x86_64-18.04-20210914_1336
docker.io/apolloauto/apollo:dev-x86_64-18.04-20210914_1336
[INFO] Remove existing Apollo Development container ...
[INFO] Determine whether host GPU is available ...
[WARNING] Cannot access GPU from within container. Please install latest Docker and NVIDIA Container Toolkit as described by: 
[WARNING]   https://github.com/NVIDIA/nvidia-docker/blob/master/README.md
[INFO] USE_GPU_HOST: 0
[INFO] Starting mounting map volumes ...

**Reply by i-kj** [reply]
**When I try to build apollo.sh I am getting this following error:**

root@in-dev-docker:/apollo# ./apollo.sh build_gpu                     
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[INFO] Apollo Environment Settings:
[INFO]     APOLLO_ROOT_DIR: /apollo
[INFO]     APOLLO_CACHE_DIR: /apollo/.cache
[INFO]     APOLLO_IN_DOCKER: true
[INFO]     APOLLO_VERSION: guardstrike/apollo_carla_bridge-2022-11-23-653b3e953
[INFO]     DOCKER_IMG: dev-x86_64-18.04-20210914_1336
[INFO]     APOLLO_ENV:  STAGE=dev USE_ESD_CAN=false
[INFO]     USE_GPU: USE_GPU_HOST=0 USE_GPU_TARGET=0
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] nvidia-smi not found. CPU will be used.
[WARNING] USE_GPU=1 without GPU can't compile. Exiting ...

**Reply by XiaoFei9704** [reply, marked as answer]
It is possible that there is a problem with your graphics card driver. You can run nvidia-smi on the local terminal to see if it is working properly. If it is not working properly, you can reinstall the graphics card driver.

**Reply by i-kj** [reply]
Thanks it worked!
